classes/author/Collector.inc.php

```diff
@@ -0,0 +1,232 @@
+
+    /** @var int */
+    public ?int $count;
+
+    /** @var int */
+    public ?int $offset;
+
+    /** @var bool|null */
+    public $includeInBrowse = null;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /**
+     * Filter by contexts
+     */
+    public function filterByContextIds(?array $contextIds): self
+    {
+        $this->contextIds = $contextIds;
+        return $this;
+    }
+
+    /**
+     * Filter by publications
+     */
+    public function filterByPublicationIds(?array $publicationIds): self
+    {
+        $this->publicationIds = $publicationIds;
+        return $this;
+    }
+
+    /**
+     * Filter by include in browse
+     */
+    public function filterByIncludeInBrowse(bool $includeInBrowse): self
+    {
+        $this->includeInBrowse = $includeInBrowse;
+        return $this;
+    }
+
+    /**
+     * Include orderBy columns to the collector query
+     */
+    public function orderBy(string $orderBy): self
+    {
+        $this->orderBy = $orderBy;
+        return $this;
+    }
+
+    /**
+     * Filter by the given and family name
+     *
+     *
+     */
+    public function filterByName(string $givenName, string $familyName): self
+    {
+        $this->givenName = $givenName;
+        $this->familyName = $familyName;
+        return $this;
+    }
+
+    /**
+     * Filter by the specified country code
+     *
+     * @param $country string Country code (2-letter)
+     *
+     * */
+    public function filterByCountry(string $country): self
+    {
+        $this->country = $country;
+        return $this;
+    }
+
+    /**
+     * Filter by the specified affiliation code
+     *
+     * */
+    public function filterByAffiliation(string $affiliation): self
+    {
+        $this->affiliation = $affiliation;
+        return $this;
+    }
+
+    /**
+     * Limit the number of objects retrieved
+     */
+    public function limit(?int $count): self
+    {
+        $this->count = $count;
+        return $this;
+    }
+
+    /**
+     * Offset the number of objects retrieved, for example to
+     * retrieve the second page of contents
+     */
+    public function offset(?int $offset): self
+    {
+        $this->offset = $offset;
+        return $this;
+    }
+
+    /**
+     * @copydoc CollectorInterface::getQueryBuilder()
+     */
+    public function getQueryBuilder(): Builder
+    {
+        $q = DB::table('authors as a')
+            ->Join('publications as p', 'a.publication_id', '=', 'p.publication_id')
+            ->Join('submissions as s', 'p.submission_id', '=', 's.submission_id');
+
+        if (isset($this->contextIds)) {
+            $q->whereIn('s.context_id', $this->contextIds);
+        }
+
+        if (isset($this->familyName) && isset($this->givenName)) {
+            $familyName = $this->familyName;
+            $givenName = $this->givenName;
+            $q->leftJoin('author_settings as fn', 'a.author_id', '=', 'fn.author_id')
```

Comment:
I'm not sure these joins are doing anything, since they're left joins -- the `fn` table alias will either match something or it won't, but the row will still be returned either way, and the `fn.*` columns are never selected. I think you'll need to convert `leftJoin` to `join` to make this filter work.

Reply:
Changed